src/main/java/uk/gov/hmcts/reform/unspec/service/CoreCaseDataService.java

```diff
@@ -27,43 +31,60 @@
     private final CoreCaseDataApi coreCaseDataApi;
     private final SystemUpdateUserConfiguration userConfig;
     private final AuthTokenGenerator authTokenGenerator;
+    private final CaseDetailsConverter caseDetailsConverter;
 
     public void triggerEvent(Long caseId, CaseEvent eventName) {
         triggerEvent(caseId, eventName, Map.of());
     }
 
     public void triggerEvent(Long caseId, CaseEvent eventName, Map<String, Object> contentModified) {
-        String userToken = idamClient.getAccessToken(userConfig.getUserName(), userConfig.getPassword());
-        String systemUpdateUserId = idamClient.getUserInfo(userToken).getUid();
+        StartEventResponse startEventResponse = startUpdate(caseId.toString(), eventName);
+        submitUpdate(caseId.toString(), caseDataContentFromStartEventResponse(startEventResponse, contentModified));
+    }
+
+    public StartEventResponse startUpdate(String caseId, CaseEvent eventName) {
+        UserDetails systemUpdateUser = getSystemUpdateUser();
 
-        StartEventResponse startEventResponse = coreCaseDataApi.startEventForCaseWorker(
-            userToken,
+        return coreCaseDataApi.startEventForCaseWorker(
+            systemUpdateUser.getUserToken(),
```

Comment:
does it make sense to hard code this to always be systemUpdate user? I can't think of a use at the moment, but this should really be generic for any user

Reply:
same problem before, so fine if you want to ignore

Reply:
If we use start using the credentials on an user (eg. solicitor, then that solicitor need to be linked with the case)